### PR TITLE
Reuse `similar` for `undef` initialization

### PR DESCRIPTION
**Pull request by goretkin**
I didn't check carefully, but I expect this is equivalent, and it looks clearer to me.

### CODE REVIEW

**Comment by goretkin** [comment]
Thanks for taking a look. I will take a look at the extra `fill`.

I don't immediately understand your first comment. I think you're talking about how this package defines methods for both `similar(A::AbstractArray, ...)`  and `similar(::Type{T}, ...) where {T<:AbstractArray}`, and that possibly there's a fallback through base so that there is just one implementation? In any case, I will take a look to see what can be done, but also wanted to ask for clarification.

File: src/OffsetArrays.jl [hunk 1]
```diff
@@ -126,15 +126,15 @@ function Base.similar(::Type{T}, shape::Tuple{OffsetAxis,Vararg{OffsetAxis}}) wh
 end
 
 Base.fill(v, inds::NTuple{N, Union{Integer, AbstractUnitRange}}) where {N} =
-    fill!(OffsetArray(Array{typeof(v), N}(undef, map(indexlength, inds)), map(indexoffset, inds)), v)
+    fill!(similar(Array{typeof(v)}, inds), v)
```
Comment (johnnychen94) [inline src/OffsetArrays.jl:129]: > I don't immediately understand your first comment.

Sorry for the inaccuracy, I was saying `similar` here actually calls this method:
 
https://github.com/JuliaArrays/OffsetArrays.jl/blob/7294f3dcab6e6117b15eea3b98ce2c2175d08194/src/OffsetArrays.jl#L123-L126 

and thus is equivalent to the previous implementation, if I intuited it correctly.

**Comment by goretkin** [comment]
I was adding a few more tests, and I see that there are some places where `===` is used, which is good since `1:3 == Base.OneTo(3)`, and some places want to test that.

But I also see many cases of e.g. `a == IdOffsetRange(3:5)` being used, and I don't see the point of doing that instead of `a == 3:5`, unless the test is really intended to check for `===`, in which case in at least a few places it should be `a === OffsetArrays.IdOffsetRange(Base.OneTo(3), 2)`.

**Comment by goretkin** [comment]
Same with `IdentityUnitRange`. I think I'll open an issue for this.

File: src/OffsetArrays.jl (hunk 1 above)
Comment (goretkin) [reply src/OffsetArrays.jl:129]: I think I just realized you might be making an observation, and not making a suggestion or a request. Is that right? Sorry again for still not understanding. :-D

**Comment by johnnychen94** [comment]
Thanks for the work @goretkin !

**Pull request merged**